### PR TITLE
DI-342 Add Dos change verification tests

### PR DESCRIPTION
**Pull request by mik3ola**
## Link to JIRA Ticket
https://nhsd-jira.digital.nhs.uk/browse/DI-342

## Description

Improve test suite with functionalities to verify individual changes sent to Dos. These changes include contacts(phone numbers and websites) and address changes.

## Type of change

- New feature (non-breaking change which adds functionality)

## Testing

Please tick the testing that has been completed

- [ ] Integration tests

## Developer Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have run the [code formatting checks](../README.md#code-quality)
- [ ] I have run the [code quality checks](../README.md#code-quality)
- [ ] New code meets [standards](https://nhsd-confluence.digital.nhs.uk/display/DI/DI+Ways+of+Working) agreed by the team
- [ ] Unit test code coverage is at or above 80%
- [ ] New and existing unit tests pass locally with my changes
- [ ] Tests have added that prove my fix is effective or that my feature works (Integration tests)
- [ ] I have made corresponding changes to the documentation
- [ ] I have cleaned down my environment (if created)

## Code Reviewer Checklist

- [ ] I have run the unit tests and they run correctly
- [ ] I can confirm the changes have been tested or approved by a tester
- [ ] I can confirm no remaining infrastructure is left over from this branch

### CODE REVIEW

File: test/integration/steps/test_parent_steps.py [hunk 1]
```diff
@@ -41,6 +42,22 @@ def a_change_event_is_valid():
     return context
 
 
+@given(parsers.parse('a Changed Event with changed "{contact}" is valid'), target_fixture="context")
+def a_changed_contact_event_is_valid(contact):
+    context = {}
+    context["change_event"] = create_change_event()
+    if contact == "website":
+        context["change_event"]["Contacts"][0]["ContactValue"] = "www.my-test-website.com"
```
Comment (ed-sparkes) [inline test/integration/steps/test_parent_steps.py:50]: The problem with this at the moment is that test will only pass the first time as a change will not be generated on subsequent runs as these values will already be committed. Instead you could have two options for each, flip between them based on current state

```
if  context["change_event"]["Contacts"][0]["ContactValue"] == "www.test1.com":
   context["change_event"]["Contacts"][0]["ContactValue"]  = "www.test2.com"
else:
   context["change_event"]["Contacts"][0]["ContactValue"]  = "www.test1.com"
```

Comment (rohith1122) [reply test/integration/steps/test_parent_steps.py:50]: may be  you can consider this as well .  you don't need to worry about index .
contact = 'website' # or Telephone
item = list(filter(lambda x:x["ContactMethodType"] == contact, change_event['Contacts']))
value = item[0]["ContactValue"]

**Review by ed-sparkes — APPROVED** [review]

**Review by JackPlowman — APPROVED** [review]
LGTM 👍

File: test/integration/steps/test_parent_steps.py (hunk 1 above)
Comment (mik3ola) [reply test/integration/steps/test_parent_steps.py]: Resolved.

**Review by JackPlowman — APPROVED** [review]
LGTM 👍

**Review by ed-sparkes — APPROVED** [review]

**Pull request merged**